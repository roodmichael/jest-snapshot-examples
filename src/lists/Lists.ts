
/**
 * Filters an array of strings
 * @param initialList 
 * @param length 
 */
export const filterByLengthGE = (initialList : Array<string>, length : number) : Array<string>  => {
   return initialList.filter((word) => word.length >= length )
}
