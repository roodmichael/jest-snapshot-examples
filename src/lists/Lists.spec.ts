import { filterByLengthGE } from './Lists';

const initialValues = [
    'Justin Trudeau',
    'Donald Trump',
    'Abraham Lincoln',
    'Winston Churchill',
    'Steven Harper',
    'Boris Johnson',
    'Rob Ford'
];

test('excludes Donald Trump and Rob Ford', () => {
    expect(filterByLengthGE(initialValues, 13)).toMatchSnapshot();
});
