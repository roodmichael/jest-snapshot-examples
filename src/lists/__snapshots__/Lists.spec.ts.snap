// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`excludes Donald Trump and Rob Ford 1`] = `
Array [
  "Justin Trudeau",
  "Abraham Lincoln",
  "Winston Churchill",
  "Steven Harper",
  "Boris Johnson",
]
`;
